src/pytest_inmanta_lsm/remote_orchestrator.py

```diff
@@ -35,80 +36,193 @@
+        """
+        Make sure that a project with the desired name or `pytest-inmanta-lsm` exists on the
+        remote orchestrator, and returns it.  Returns the project after the configuration has
+        been applied.
+
+        :param client: The client which can reach the orchestrator we want to configure.
+        """
+        project_name = self.project or "pytest-inmanta-lsm"
+
+        for raw_project in client.project_list().result["data"]:
+            project = inmanta.data.model.Project(**raw_project)
+            if project.name == project_name:
+                return project
+
+        # We didn't find any project with the desired name, so we create a new one
+        result = client.project_create(name=project_name)
+        assert result.code in range(200, 300), str(result.result)
+
+        return inmanta.data.model.Project(**result.result["data"])
+
+    def configure_environment(self, client: inmanta.protocol.endpoints.SyncClient) -> inmanta.data.model.Environment:
+        """
+        Make sure that our environment exists on the remote orchestrator, and has the desired
+        name and project.  Returns the environment after the configuration has been applied.
+
+        :param client: The client which can reach the orchestrator we want to configure.
+        """
+        try:
+            current_environment = self.get_environment(client)
+        except LookupError:
+            # The environment doesn't exist, we create it
+            result = client.environment_create(
+                environment_id=self.id,
+                name=self.name or "pytest-inmanta-lsm",
+                project_id=self.configure_project(client).id,
+            )
+            assert result.code in range(200, 300), str(result.result)
+            return inmanta.data.model.Environment(**result.result["data"])
+
+        current_project = self.get_project(client)
+
+        updates: dict[str, object] = dict()
+        if self.name is not None and current_environment.name != self.name:
+            # We care about the environment name is it is not a match
+            # We update the environment name
+            updates["name"] = self.name
+
+        if self.project is not None and current_project.name != self.name:
+            # We care about the project name and it is not a match
+            # We make sure the project with the desired name exists and
+            # assign our environment to it
+            updates["project_id"] = self.configure_project(client).id
+
+        if len(updates) > 0:
+            # Apply the updates
+            result = client.environment_modify(
+                id=self.id,
+                **updates,
+            )
+
+            assert result.code in range(200, 300), str(result.result)
+            return inmanta.data.model.Environment(**result.result["data"])
+        else:
+            return current_environment
+
+    def __str__(self) -> str:
+        return str(self.id)
```

Comment:
You are right, I tried to gain some time because I didn't want to update the usage of the environment attribute everywhere, but I think a better solution is simply to keep the old environment attribute and add a new one that contains our new object.